airflow/providers/amazon/aws/hooks/athena_sql.py

```diff
@@ -0,0 +1,104 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+import pyathena
+from sqlalchemy.engine.url import URL
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.common.sql.hooks.sql import DbApiHook
+
+if TYPE_CHECKING:
+    from pyathena.connection import Connection as AthenaConnection
+
+
+class AthenaSQLHook(AwsBaseHook, DbApiHook):
+    """Interact with Amazon Athena.
+
+    Provide wrapper around PyAthena library.
+
+    :param athena_conn_id: :ref:`Amazon Athena Connection <howto/connection:athena>`.
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    In case of passing ``aws_conn_id`` the authentication parameters will be retrieved from that connection, otherwise ``athena_conn_id`` will be used.
+
+    PyAthena and AWS Authentication parameters could be passed in extra field of ``athena_conn_id`` connection.
+
+    .. note::
+        get_uri() depends on SQLAlchemy and PyAthena.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    conn_name_attr = "athena_conn_id"
+    default_conn_name = "athena_default"
+    conn_type = "athena"
+    hook_name = "Amazon Athena"
+    supports_autocommit = True
+
+    def __init__(self, athena_conn_id: str | None = default_conn_name, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.athena_conn_id = athena_conn_id
+        if "aws_conn_id" not in kwargs:
+            self.aws_conn_id = athena_conn_id
+
+    @cached_property
+    def conn(self):
+        """Get athena connection object."""
+        return self.get_connection(self.athena_conn_id)
+
+    def _get_conn_params(self) -> dict[str, str | None]:
+        """Retrieve connection parameters."""
+        return dict(
+            driver=self.conn.extra_dejson.get("driver", "rest"),
+            schema_name=self.conn.schema,
+            region_name=self.region_name,
+        )
+
+    def get_uri(self) -> str:
+        """Overridden to use the Athena dialect as driver name."""
+        conn_params = self._get_conn_params()
+        creds = self.get_credentials(region_name=conn_params["region_name"])
+
+        return URL.create(
+            f'awsathena+{conn_params["driver"]}',
+            username=creds.access_key,
+            password=creds.secret_key,
+            host=f'athena.{conn_params["region_name"]}.amazonaws.com',
```

Comment:
What about other AWS partitions, such as `aws-cn` and `aws-us-gov` I guess it might not work with regions, outside of `aws` partition, such as:
- `cn-north-1` (China)
- `us-gov-east-1` (US Goverment)

Reply:
good point! working on that!